src/main/java/com/commercetools/sync/commons/utils/GenericUpdateActionUtils.java

```diff
@@ -84,21 +84,21 @@
      * @param <T>              the type of the resource to do the update action on.
      * @param syncOptions      responsible for supplying the sync options to the sync utility method.
      * @return a setCustomField update action on the provided field name, with the provided value
-     * on the resource it's requested on.
+     *      on the resource it's requested on.
      */
     @Nonnull
     static <T extends Custom & Resource<T>> Optional<UpdateAction<T>> buildTypedSetCustomFieldUpdateAction(
-            @Nonnull final String customFieldName,
-            @Nullable final JsonNode customFieldValue,
-            @Nonnull final T resource,
-            @Nonnull final BaseSyncOptions syncOptions) {
+        @Nonnull final String customFieldName,
+        @Nullable final JsonNode customFieldValue,
+        @Nonnull final T resource,
+        @Nonnull final BaseSyncOptions syncOptions) {
         try {
-            return Optional.of(GenericCustomActionBuilderFactory
-                    .of(resource)
-                    .buildSetCustomFieldAction(customFieldName, customFieldValue));
-        } catch (BuildUpdateActionException | IllegalAccessException | InstantiationException e) {
+            return Optional.of(GenericCustomActionBuilderFactory.of(resource)
+                                                                .buildSetCustomFieldAction(customFieldName,
+                                                                    customFieldValue));
+        } catch (BuildUpdateActionException | IllegalAccessException | InstantiationException exception) {
             syncOptions.applyErrorCallback(format(SET_CUSTOM_FIELD_BUILD_FAILED.getDescription(), customFieldName,
-                    resource.toReference().getTypeId(), resource.getId(), e.getMessage()), e);
+                resource.toReference().getTypeId(), resource.getId(), exception.getMessage()), exception);
```

Comment:
- In this case, it might be not useful to send same message of exception, because as you said, he can already get it from the exception itself. But in other cases, we might need to pass more custom messages other than the ones in the exception itself. 

- Well, it gives the user more flexibility on how to handle failures according to each exception. Instead of just getting a message.

-nope, because this reference is initialised internally in the JVM SDK for every CTP resource.